src/videocontext.js

```diff
@@ -957,7 +964,23 @@ export default class VideoContext {
             for (let node of sortedNodes) {
                 if (this._sourceNodes.indexOf(node) === -1) {
                     node._update(this._currentTime);
-                    node._render();
+                    // to avoid hight GPU/Memory usage and improve performance
+                    // we should only render when video is playing or if current time has changed
+                    if (this._state === VideoContext.STATE.PLAYING) {
+                        node._render();
+                    } else {
+                        const ready = !this._isStalled();
+                        // if nodes are ready and current time has changed
+                        // allow 6 frames to render new changes
+                        if (ready && this._prevTime !== this._currentTime) {
+                            this._renderTimes = 6;
```

Comment:
Why 6 ? Also this does break my overlay rendering as my call to update after rendering into the CanvasNode now no-ops... I wonder if we could add some sort of logic to tell if the render graph has changed, and then also add a forced render method? 

In my use case I haven't changed the render graph, but I have drawn into a node.  How about this idea?

- Add isDirty flag to nodes.
We update this method to do the following:
- If playing, always render -- although we should also throttle this to the actual video fps at some point -- not for now
- Otherwise, if the node graph structure has changed -> render
- Otherwise if any parent nodes of the node we're about to render is marked as dirty -> render (and mark the node us no longer dirty)

Then in my use case (and in future ones, we can do something to a node, mark it dirty and have it re-render in the next animation frame)

Reply:
After the node is ready the first render doesn't show the frame on the canvas, doing a couple of renders will work, that 6 it was just to make sure it will render the frame but I don't know the reason why that doesn't work with only one render..
That is a good idea, I will work on that approach, Thanks!

Reply:
updated the code to use `isDirty` flag and render only once after nodes are ready, also added `renderOnDirtyNodeOnly` config flag to update this feature